src/main/java/pl/put/poznan/transformer/texttransformers/ReverseTextTransformer.java

```diff
@@ -0,0 +1,22 @@
+package pl.put.poznan.transformer.texttransformers;
+
+/**
+ * Reverse String but preserve the capitalization of letters (indexes of capitalized letters don't change).
+ *
+ * @author Michał Grygiel
+ * @see TextTransformer
+ */
+public class ReverseTextTransformer implements TextTransformer {
+    @Override
+    public String transform(String text) {
+        StringBuilder builder = new StringBuilder(text);
+        builder.reverse();
+        for (int i = 0; i < text.length(); i++) {
+            if (Character.isUpperCase(text.charAt(i)))
+                builder.setCharAt(i, Character.toUpperCase(builder.charAt(i)));
+            else
+                builder.setCharAt(i, Character.toLowerCase(builder.charAt(i)));
+        }
+        return builder.toString();
+    }
+}
```

Comment:
klasa musi mieć adnotacje `@Component`, aby `TextTransformerService` ją wykrył

Reply:
Dodałem commita do brancha który to naprawia